### PR TITLE
chore: tooling improvements

### PR DESCRIPTION
**Pull request by IrakliJani**
followup of https://github.com/lightdash/lightdash/pull/2969 ← closing

resolves: https://github.com/lightdash/lightdash/issues/2968

- extracts all tooling packages to root package.json and removes version differences
- upgrades TS to latest version
- enables `"useUnknownInCatchVariables: true"` for typescript to avoid compile errors (or massive migration)
- normalizes ts config and extends ts config from root
- normalizes eslint config and extends eslint config from root
- normalizes prettier config and extends prettier config from root
- adds fix-lint and fix-format to all packages
- fixes jest for frontend (without CRA magic)
- fixes editor problems with ESlint config (mine was half broken/half working)
- brings back crucial react specific eslint rules (missing keys, exhaustive hook deps). some of them are disabled now but we can enable them in following PRs.
- fixes https://github.com/lightdash/lightdash/issues/1578
- removes webpack / adds vite

@lightdash/engineering give it a try and let me know. For me it is a huge improvement in terms of speed. But if you don't like vite/its slower than what we had before I'll use rollup/webpack on top of tooling changes

right now - locally everything works fine
- [x] lints
- [x] formatter
- [x] tests
- [x] e2e
- [x] dev
- [x] build
- [x] start

need to test:
- [x] deploy checks
- [x] ~~source maps~~

### CODE REVIEW

File: packages/backend/src/index.ts [hunk 1]
```diff
@@ -173,10 +173,6 @@ if (
 // frontend
 app.use(express.static(path.join(__dirname, '../../frontend/build')));
 
-app.get('*', (req, res) => {
-    res.sendFile(path.join(__dirname, '../../frontend/build', 'index.html'));
-});
-
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/backend/src/index.ts:179]: @ZeRego frontend should be running at https://localhost:3000

[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/backend/src/index.ts:179]: *http

[the remark this answers is not in the document]
Comment (ZeRego) [reply packages/backend/src/index.ts:179]: I used `yarn build` and then `yarn start` to run in production mode.

[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/backend/src/index.ts:179]: ah I see.. I'll update with a proper path

**Comment by ZeRego** [comment]
fyi: I tried enabling [source map](https://vitejs.dev/config/build-options.html#build-sourcemap) and got a `FATAL ERROR: MarkCompactCollector: young object promotion failed Allocation failed - JavaScript heap out of memory` 😂

**Comment by IrakliJani** [comment]
@ZeRego I get an error from render blueprints.

https://dashboard.render.com/web/srv-cci4jnta4995109jbi30/deploys/dep-ccr9749a6gdmbse86jog?mode=maximize

```Sep 30 11:01:21 AM  #22 2.595 error TS5083: Cannot read file '/usr/app/tsconfig.json'.```

however I updated `dockerfile` to `COPY tsconfig.json .`

any ideas?

**Comment by ZeRego** [comment]
@IrakliJani pushed a fix for the pegjs import that was preventing the app to run. 
Also pushed a potential fix for the out of memory by chuncking manually the output. This approach is a bit over the top but was good to see what libs are the heaviest. Maybe we can just chunk those or optimize the imports.

Screenshot 2022-10-10 at 17 45 35

**Comment by IrakliJani** [comment]
@ZeRego updated PR. one problem with current chunking is this: 

```
<script type="module" crossorigin src="/assets/index.29da37a3.js"></script>
<link rel="modulepreload" crossorigin href="/assets/object-assign.5efa5c82.js">
<link rel="modulepreload" crossorigin href="/assets/regenerator-runtime.ed378335.js">
<link rel="modulepreload" crossorigin href="/assets/core-js-pure.7b85189e.js">
<link rel="modulepreload" crossorigin href="/assets/@babel.23046e1f.js">
<link rel="modulepreload" crossorigin href="/assets/react.b4f4cedd.js">
<link rel="modulepreload" crossorigin href="/assets/react-dom.f87dc6d8.js">
<link rel="modulepreload" crossorigin href="/assets/classnames.49ffb9ee.js">
<link rel="modulepreload" crossorigin href="/assets/react-day-picker.a2c5162c.js">
<link rel="modulepreload" crossorigin href="/assets/date-fns.4cedfee7.js">
<link rel="modulepreload" crossorigin href="/assets/lodash-es.5186fbe9.js">
<link rel="modulepreload" crossorigin href="/assets/date-fns-tz.e0beca5b.js">
<link rel="modulepreload" crossorigin href="/assets/tslib.f0242f4b.js">
<link rel="modulepreload" crossorigin href="/assets/lower-case.7b45055c.js">
<link rel="modulepreload" crossorigin href="/assets/no-case.b5d95c2f.js">
<link rel="modulepreload" crossorigin href="/assets/pascal-case.1771dc6f.js">
<link rel="modulepreload" crossorigin href="/assets/react-fast-compare.983ab98a.js">
<link rel="modulepreload" crossorigin href="/assets/@popperjs.d582507c.js">
<link rel="modulepreload" crossorigin href="/assets/@juggle.cbd8e6c0.js">
<link rel="modulepreload" crossorigin href="/assets/prop-types.007f5878.js">
<link rel="modulepreload" crossorigin href="/assets/dom-helpers.e11a883e.js">
<link rel="modulepreload" crossorigin href="/assets/react-transition-group.ca3ddc47.js">
<link rel="modulepreload" crossorigin href="/assets/object-keys.c978a89e.js">
<link rel="modulepreload" crossorigin href="/assets/define-properties.8720f8fc.js">
<link rel="modulepreload" crossorigin href="/assets/has-symbols.bc147734.js">
<link rel="modulepreload" crossorigin href="/assets/function-bind.2e53367c.js">
<link rel="modulepreload" crossorigin href="/assets/has.823e354e.js">
<link rel="modulepreload" crossorigin href="/assets/get-intrinsic.f642d647.js">
<link rel="modulepreload" crossorigin href="/assets/call-bind.14ba4059.js">
<link rel="modulepreload" crossorigin href="/assets/is-arguments.2fa155a1.js">
<link rel="modulepreload" crossorigin href="/assets/object-is.1c2a929c.js">
<link rel="modulepreload" crossorigin href="/assets/is-date-object.18759a20.js">
<link rel="modulepreload" crossorigin href="/assets/deep-equal.d5aefebe.js">
<link rel="modulepreload" crossorigin href="/assets/popper.js.ac7712bb.js">
<link rel="modulepreload" crossorigin href="/assets/warning.ca260113.js">
<link rel="modulepreload" crossorigin href="/assets/gud.36d504fc.js">
<link rel="modulepreload" crossorigin href="/assets/@hypnosphi.4ed02eae.js">
<link rel="modulepreload" crossorigin href="/assets/react-popper.d191c6b0.js">
<link rel="modulepreload" crossorigin href="/assets/@blueprintjs.2dfecf23.js">
<link rel="modulepreload" crossorigin href="/assets/react-query.f94beb91.js">
<link rel="modulepreload" crossorigin href="/assets/resolve-pathname.15ab1017.js">
<link rel="modulepreload" crossorigin href="/assets/value-equal.dd886cd8.js">
<link rel="modulepreload" crossorigin href="/assets/tiny-invariant.251df24d.js">
<link rel="modulepreload" crossorigin href="/assets/history.5f9270eb.js">
<link rel="modulepreload" crossorigin href="/assets/mini-create-react-context.d9dbb0ee.js">
<link rel="modulepreload" crossorigin href="/assets/isarray.1e9043d8.js">
<link rel="modulepreload" crossorigin href="/assets/react-is.6e3a4316.js">
<link rel="modulepreload" crossorigin href="/assets/hoist-non-react-statics.ca2696ea.js">
<link rel="modulepreload" crossorigin href="/assets/react-router.bb2c052b.js">
<link rel="modulepreload" crossorigin href="/assets/react-router-dom.0bd32984.js">
<link rel="modulepreload" crossorigin href="/assets/@emotion.33661adb.js">
<link rel="modulepreload" crossorigin href="/assets/styled-components.9ce31169.js">
<link rel="modulepreload" crossorigin href="/assets/moment.849c9289.js">
<link rel="modulepreload" crossorigin href="/assets/dependency-graph.6c9a2ab5.js">
<link rel="modulepreload" crossorigin href="/assets/pegjs.f4b4f890.js">
<link rel="modulepreload" crossorigin href="/assets/uuid.e6310df4.js">
<link rel="modulepreload" crossorigin href="/assets/liquidjs.55e43324.js">
<link rel="modulepreload" crossorigin href="/assets/@ucast.67a68393.js">
<link rel="modulepreload" crossorigin href="/assets/@casl.bc609bd8.js">
<link rel="modulepreload" crossorigin href="/assets/react-use-intercom.535234b0.js">
<link rel="modulepreload" crossorigin href="/assets/cohere-js.211f21fd.js">
<link rel="modulepreload" crossorigin href="/assets/@sentry.4d44ddb4.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-chunked.89685333.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-combine-extensions.273ad418.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-character.34fe9ef6.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-extension-gfm-autolink-literal.65ed901e.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-factory-space.9ceef495.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-normalize-identifier.1b90ffa2.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-classify-character.b0b2c212.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-resolve-all.9c5f4c34.js">
<link rel="modulepreload" crossorigin href="/assets/parse-entities.f90f9b8a.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-subtokenize.6aea0fac.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-factory-destination.6387c6ce.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-factory-label.75902521.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-factory-title.354b860a.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-factory-whitespace.51cb8e03.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-html-tag-name.50973316.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-core-commonmark.377e1992.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-extension-gfm-footnote.4469113a.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-extension-gfm-strikethrough.5fab04ea.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-extension-gfm-table.0edecbce.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-extension-gfm-task-list-item.e9d9fc89.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-extension-gfm.0bc1c84a.js">
<link rel="modulepreload" crossorigin href="/assets/ccount.437a91fc.js">
<link rel="modulepreload" crossorigin href="/assets/unist-util-is.aa7a6b8f.js">
<link rel="modulepreload" crossorigin href="/assets/unist-util-visit-parents.a8094f4c.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-find-and-replace.47b6cc48.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-gfm-autolink-literal.6e1887dc.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-decode-numeric-character-reference.8339c51c.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-decode-string.bb622224.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-to-markdown.693a2872.js">
<link rel="modulepreload" crossorigin href="/assets/unist-util-visit.18b91973.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-gfm-footnote.a6624ea3.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-gfm-strikethrough.eb3c052d.js">
<link rel="modulepreload" crossorigin href="/assets/markdown-table.1d4da583.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-gfm-table.62a27fed.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-gfm-task-list-item.63109306.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-gfm.068b700e.js">
<link rel="modulepreload" crossorigin href="/assets/remark-gfm.1ca5b8b9.js">
<link rel="modulepreload" crossorigin href="/assets/github-slugger.91cf43d2.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-has-property.644e6498.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-heading-rank.0a80fc76.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-to-string.1ad81310.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-slug.b9ce94b6.js">
<link rel="modulepreload" crossorigin href="/assets/parse5.76e47094.js">
<link rel="modulepreload" crossorigin href="/assets/unist-util-position.0143b5fd.js">
<link rel="modulepreload" crossorigin href="/assets/html-void-elements.7f9a8f3d.js">
<link rel="modulepreload" crossorigin href="/assets/web-namespaces.6ad0c8b2.js">
<link rel="modulepreload" crossorigin href="/assets/zwitch.b2c51887.js">
<link rel="modulepreload" crossorigin href="/assets/property-information.3146165c.js">
<link rel="modulepreload" crossorigin href="/assets/vfile-location.50cd4731.js">
<link rel="modulepreload" crossorigin href="/assets/space-separated-tokens.51b2c0f2.js">
<link rel="modulepreload" crossorigin href="/assets/comma-separated-tokens.a7c16398.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-from-parse5.dee2e2ea.js">
<link rel="modulepreload" crossorigin href="/assets/inline-style-parser.e25dc57b.js">
<link rel="modulepreload" crossorigin href="/assets/style-to-object.ac23b492.js">
<link rel="modulepreload" crossorigin href="/assets/hast-to-hyperscript.40a6e41a.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-to-parse5.c78fcd26.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-raw.b239bea5.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-raw.1c246c96.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-attr.0a686143.js">
<link rel="modulepreload" crossorigin href="/assets/xtend.b45ce5b8.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-parse-selector.d8f3d68d.js">
<link rel="modulepreload" crossorigin href="/assets/hastscript.8b658e01.js">
<link rel="modulepreload" crossorigin href="/assets/is-decimal.49dd509c.js">
<link rel="modulepreload" crossorigin href="/assets/prismjs.6c07b686.js">
<link rel="modulepreload" crossorigin href="/assets/@mapbox.8fe402b3.js">
<link rel="modulepreload" crossorigin href="/assets/direction.bdd2dece.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-is-element.784e9380.js">
<link rel="modulepreload" crossorigin href="/assets/bcp-47-match.a43c2be3.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-whitespace.7ef6e8a4.js">
<link rel="modulepreload" crossorigin href="/assets/css-selector-parser.a871a66f.js">
<link rel="modulepreload" crossorigin href="/assets/boolbase.c1acfbca.js">
<link rel="modulepreload" crossorigin href="/assets/nth-check.8996ace7.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-select.14089943.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-rewrite.add755cc.js">
<link rel="modulepreload" crossorigin href="/assets/extend.e5b6ae6f.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-autolink-headings.49a5b84a.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-to-string.8578f5ea.js">
<link rel="modulepreload" crossorigin href="/assets/micromark.31bf47fc.js">
<link rel="modulepreload" crossorigin href="/assets/unist-util-stringify-position.b30cfc55.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-from-markdown.7ebb573d.js">
<link rel="modulepreload" crossorigin href="/assets/remark-parse.cbd2869b.js">
<link rel="modulepreload" crossorigin href="/assets/bail.a36530c7.js">
<link rel="modulepreload" crossorigin href="/assets/is-plain-obj.52d91e44.js">
<link rel="modulepreload" crossorigin href="/assets/trough.90fb8a5f.js">
<link rel="modulepreload" crossorigin href="/assets/vfile-message.ad2f416b.js">
<link rel="modulepreload" crossorigin href="/assets/vfile.2d27a9c9.js">
<link rel="modulepreload" crossorigin href="/assets/unified.3ed68b50.js">
<link rel="modulepreload" crossorigin href="/assets/unist-builder.bfc351d0.js">
<link rel="modulepreload" crossorigin href="/assets/unist-util-generated.9a6cceee.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-definitions.99b7a9b6.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-encode.4a3c6206.js">
<link rel="modulepreload" crossorigin href="/assets/micromark-util-sanitize-uri.8810fce8.js">
<link rel="modulepreload" crossorigin href="/assets/mdurl.c429b110.js">
<link rel="modulepreload" crossorigin href="/assets/mdast-util-to-hast.d89685cb.js">
<link rel="modulepreload" crossorigin href="/assets/remark-rehype.0b79beec.js">
<link rel="modulepreload" crossorigin href="/assets/react-markdown.3a47db9e.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-parse.51d8e067.js">
<link rel="modulepreload" crossorigin href="/assets/character-entities-legacy.6c26a43c.js">
<link rel="modulepreload" crossorigin href="/assets/character-entities-html4.4bfa2010.js">
<link rel="modulepreload" crossorigin href="/assets/stringify-entities.fe71fcae.js">
<link rel="modulepreload" crossorigin href="/assets/hast-util-to-html.82872ef0.js">
<link rel="modulepreload" crossorigin href="/assets/rehype-stringify.ff6121b8.js">
<link rel="modulepreload" crossorigin href="/assets/rehype.90bc6eb6.js">
<link rel="modulepreload" crossorigin href="/assets/@uiw.49e47c4c.js">
<link rel="modulepreload" crossorigin href="/assets/rudder-sdk-js.29fff740.js">
<link rel="modulepreload" crossorigin href="/assets/immer.05c277d6.js">
<link rel="modulepreload" crossorigin href="/assets/scheduler.b32e8061.js">
<link rel="modulepreload" crossorigin href="/assets/use-context-selector.62f31abc.js">
<link rel="modulepreload" crossorigin href="/assets/@tanstack.2090a4eb.js">
<link rel="modulepreload" crossorigin href="/assets/fuse.js.7ec25fd3.js">
<link rel="modulepreload" crossorigin href="/assets/fast-deep-equal.c393d0d2.js">
<link rel="modulepreload" crossorigin href="/assets/react-use.fd04f9a3.js">
<link rel="modulepreload" crossorigin href="/assets/react-hook-form.e0f6381a.js">
<link rel="modulepreload" crossorigin href="/assets/@hookform.d1a41514.js">
<link rel="modulepreload" crossorigin href="/assets/toggle-selection.465a7f22.js">
<link rel="modulepreload" crossorigin href="/assets/copy-to-clipboard.baf92849.js">
<link rel="modulepreload" crossorigin href="/assets/react-copy-to-clipboard.df200522.js">
<link rel="modulepreload" crossorigin href="/assets/lodash.isequal.869c0378.js">
<link rel="modulepreload" crossorigin href="/assets/clsx.26dd6e97.js">
<link rel="modulepreload" crossorigin href="/assets/react-draggable.b950f4b2.js">
<link rel="modulepreload" crossorigin href="/assets/react-resizable.ae0ccd1a.js">
<link rel="modulepreload" crossorigin href="/assets/react-grid-layout.6409092a.js">
<link rel="modulepreload" crossorigin href="/assets/react-csv.288d601e.js">
<link rel="modulepreload" crossorigin href="/assets/zrender.0289a767.js">
<link rel="modulepreload" crossorigin href="/assets/echarts.ad2dfb10.js">
<link rel="modulepreload" crossorigin href="/assets/size-sensor.dd576941.js">
<link rel="modulepreload" crossorigin href="/assets/echarts-for-react.a3464d80.js">
<link rel="modulepreload" crossorigin href="/assets/redux.1440a872.js">
<link rel="modulepreload" crossorigin href="/assets/react-redux.1ededf27.js">
<link rel="modulepreload" crossorigin href="/assets/use-memo-one.0feb0597.js">
<link rel="modulepreload" crossorigin href="/assets/css-box-model.b2ef854e.js">
<link rel="modulepreload" crossorigin href="/assets/memoize-one.a4c8bbc0.js">
<link rel="modulepreload" crossorigin href="/assets/raf-schd.1e27b7ec.js">
<link rel="modulepreload" crossorigin href="/assets/react-beautiful-dnd.78be948d.js">
<link rel="modulepreload" crossorigin href="/assets/ace-builds.3ab56247.js">
<link rel="modulepreload" crossorigin href="/assets/lodash.get.cf51a3ae.js">
<link rel="modulepreload" crossorigin href="/assets/diff-match-patch.f318f9bb.js">
<link rel="modulepreload" crossorigin href="/assets/react-ace.5326a7fa.js">
<link rel="modulepreload" crossorigin href="/assets/tinycolor2.f9a54f2b.js">
<link rel="modulepreload" crossorigin href="/assets/lodash.5cecd36f.js">
<link rel="modulepreload" crossorigin href="/assets/reactcss.9aa5ae61.js">
<link rel="modulepreload" crossorigin href="/assets/material-colors.0d188cc0.js">
<link rel="modulepreload" crossorigin href="/assets/@icons.9c00e9a5.js">
<link rel="modulepreload" crossorigin href="/assets/react-color.ea727c0f.js">
<link rel="modulepreload" crossorigin href="/assets/fflate.cfd4678a.js">
<link rel="modulepreload" crossorigin href="/assets/jspdf.26e5c8e6.js">
<link rel="modulepreload" crossorigin href="/assets/highlight.js.006865ff.js">
<link rel="modulepreload" crossorigin href="/assets/format.2f4c8e09.js">
<link rel="modulepreload" crossorigin href="/assets/fault.108adcb5.js">
<link rel="modulepreload" crossorigin href="/assets/lowlight.f5ebc2a8.js">
<link rel="modulepreload" crossorigin href="/assets/react-syntax-highlighter.c8e52135.js">
<link rel="stylesheet" href="/assets/@blueprintjs.3e63b75e.css">
<link rel="stylesheet" href="/assets/@uiw.0f0ce91f.css">
<link rel="stylesheet" href="/assets/index.5e3cfb18.css">
```

I will compare new load times with the ones from webpack

if it's gotten worse I'll try to stack up smaller chunks and bundle blueprint, react-syntax-highlighter and stuff like that separately

**Review by ZeRego — APPROVED** [review]
Lets keeps sourcemaps out of this PR and merge it 👍

**Pull request merged**